runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowRunner.java

```diff
@@ -1365,21 +1365,19 @@ void addPCollectionRequiringIndexedFormat(PCollection<?> pcol) {
   }
 
   void maybeRecordPCollectionWithAutoSharding(PCollection<?> pcol) {
-    if (hasExperiment(options, "beam_fn_api")) {
+    if (!options.isEnableStreamingEngine()) {
       LOG.warn(
-          "Runner determined sharding not available in Dataflow for GroupIntoBatches for portable "
-              + "jobs. Default sharding will be applied.");
+          "Runner determined sharding not available in Dataflow for GroupIntoBatches for "
+              + "non-Streaming Engine jobs. Default sharding will be applied.");
       return;
     }
-    if (!options.isEnableStreamingEngine()) {
+    if (!hasExperiment(options, "enable_streaming_auto_sharding")) {
       LOG.warn(
-          "Runner determined sharding not available in Dataflow for GroupIntoBatches for Streaming "
-              + "Appliance jobs. Default sharding will be applied.");
+          "Runner determined sharding not enabled in Dataflow for GroupIntoBatches for Streaming "
+              + "Engine jobs. Default sharding will be applied.");
       return;
     }
-    if (hasExperiment(options, "enable_streaming_auto_sharding")) {
-      pcollectionsRequiringAutoSharding.add(pcol);
-    }
+    pcollectionsRequiringAutoSharding.add(pcol);
```

Comment:
I think we also want to want to check the case `hasExperiment(options, "beam_fn_api") && !hasExperiment(options, "use_runner_v2")`

Reply:
Good catch. Added the check.